_posts/2023-08-19-kafka-node-pools-node-id-management.md

```diff
@@ -0,0 +1,265 @@
+It will instead remove the Kafka node with the highest node IDs from the second pool which was scaled down which are the nodes 4 and 5.
+
+Technically, this works perfectly fine.
+But it can easily get a bit confusing when managing and monitoring the Kafka cluster.
+After the scale up and scale down, out cluster now has node IDs 0, 1, 2, 3, 6, and 7.
+And out of that, the node IDs 0, 1, 2, 6, and 7 belong to the first node pool.
+And the node ID 3 belongs to the second node pool.
+
+This can get easily confusing.
+Imagine your Kafka cluster is running for several years and there were many different changes, scale-ups and scale-downs over the period.
+When someone calls you in the middle of the night about a production issue on node 6, will you remember to which node pool does it belong to?
+Or what configuration does it use?
+We thought about this and we have prepared a way how you can assign different range of node IDs for each node pool.
+
+### Assigning node ID ranges to node pools
+
+You can use the `strimzi.io/next-node-ids` annotation on the `KafkaNodePool` resource to tell Strimzi what node IDs should be used when a new node is being added to the node pool.
+The annotation can contain one or more IDs or ID ranges.
+It looks like this:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaNodePool
+metadata:
+  name: pool-b
+  labels:
+    strimzi.io/cluster: my-cluster
+  annotations:
+    strimzi.io/next-node-ids: "[100-199]"
+spec:
+  # ...
+```
+
+It has no impact on any nodes and their node IDs which might already exist in the node pool.
+But when a new node is created, Strimzi will find the lowest available node ID from this range and use it.
+In case all node IDs from the range would be already in-use, Strimzi will use the lowest available node ID outside of this range.
+If you want, you can also combine multiple ranges or use individual node IDs: `"[5, 8, 10-15, 18, 20-25]"`.
+
+Let's imagine the same situation as we discussed in the previous section, but this time with the two node pools using their own node ID ranges:
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaNodePool
+metadata:
+  name: pool-a
+  labels:
+    strimzi.io/cluster: my-cluster
+  annotations:
+    strimzi.io/next-node-ids: "[0-99]"
+spec:
+  # ...
+---
+
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaNodePool
+metadata:
+  name: pool-b
+  labels:
+    strimzi.io/cluster: my-cluster
+  annotations:
+    strimzi.io/next-node-ids: "[100-199]"
+spec:
+  # ...
+```
+
+When they are first deployed, the node IDs in the first pool will be 0, 1, and 2.
+And the node IDs in the second pool will be 100, 101, and 102.
+After the scale up and scale down, the first node pool will use node IDs 0, 1, 2, 3 and 4.
+And the second node pool will use node ID 100.
+Since each node pool has its own range of node IDs, they do not mix and it is much easier to see which node pool does the node ID belong to.
```

Comment:
```suggestion
Since each node pool has its own range of node IDs, they do not mix and it is much easier to see which node pool the node ID belongs to.
```